project-api-tests/src/test/java/org/ballerina/projectapi/CentralTestUtils.java

```diff
@@ -199,4 +229,148 @@ static Path getBalaPath(Path projectPath, String org, String pkgName, String pla
         return projectPath.resolve("target").resolve("bala")
                 .resolve(org + "-" + pkgName + "-" + platform + "-" + version + ".bala");
     }
+
+    /**
+     * Check if package already available on central.
+     *
+     * @param pkg package
+     * @param tempWorkspaceDirectory Path to workspace
+     * @param envVariables Environmental variables
+     * @return whether the package is available on central
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    static boolean isPkgAvailableInCentral(String pkg, Path tempWorkspaceDirectory,
+                                           Map<String, String> envVariables) throws IOException, InterruptedException {
+        String buildOutput = searchPackageDetails(pkg, tempWorkspaceDirectory, envVariables);
+        return !buildOutput.contains("no modules found");
+    }
+
+    /**
+     * Returns true if the provided version is available on central.
+     *
+     * @param pkg package
+     * @param tempWorkspaceDirectory Path to workspace
+     * @param envVariables Environmental variables
+     * @param version version to check availability
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    static boolean isPkgVersionAvailableInCentral(String pkg, Path tempWorkspaceDirectory,
+                                                  Map<String, String> envVariables, String version)
+            throws IOException, InterruptedException {
+        String buildOutput = searchPackageDetails(pkg, tempWorkspaceDirectory, envVariables);
+        return (!buildOutput.contains("no modules found") && buildOutput.contains(version));
+    }
+
+    /**
+     * Search for the given package on Central.
+     *
+     * @param pkg package
+     * @param tempWorkspaceDirectory Path to workspace
+     * @param envVariables Environmental variables
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    private static String searchPackageDetails(String pkg, Path tempWorkspaceDirectory,
```

Comment:
Won't this be useful for other tests as well? If so, we can make it package-private/public

Reply:
This method contains reused logic used by other package-private/public methods within the same class. So, IMO no need to make this available.